test/gpu/native/basics/outerRefFuture.chpl

```diff
@@ -5,12 +5,10 @@ on here.gpus[0] {
 
   ref xRef = x;
 
-  // as of today, we don't have a good implementation for shadow
-  // variables/intents for foreach. But this test exposes a case where the
-  // behavior is wrong. We'd expect this code to change x to 2. But as we
-  // offload x and use a GPU-only ref in the kernel, you don't see any changes
-  // made to x.
-  foreach a in A do xRef = a;
+  // we're not processing 'ref' intents of scalars for GPUs properly.  We'd
+  // expect this code to change x to 2. But as we offload x and use a GPU-only
+  // ref in the kernel, you don't see any changes made to x.
+  foreach a in A with (ref xRef) do xRef = a;
```

Comment:
I wonder what the current behavior here is -- I think it is unchanged? The "right" behavior today is to make this cause communication, _and_ have that cause a runtime failure because the GPU kernel is inside a `local` block and can't initiate comm. Currently, potentially communication operations like that behave that way inside `local` blocks in Chapel. But we may be narrowing things down to prevent that from happening. For one, I think we keep arguments to the kernel launch primitive narrow no matter what. I would imagine changing that could be one of the first PRs leading to GPU-driven communication and would make this case give a runtime error, which is a more desirable behavior then it is today.

That being said, do you think there could be any shorter-term solution to this one here. Can we:

- make the compiler error/thwart kernel generation with `ref` intents on scalars?
- maybe do that in the runtime where we can actually check if something has a ref intent _and_ not allocated on device memory?

(Not suggesting it should be part of this PR, but it can be a low effort thing that can improve the situation until we add GPU-driven comm)

Reply:
> I wonder what the current behavior here is -- I think it is unchanged?

I believe so. Basically for now making anything `ref` will just keep the old behavior. I also didn't change `outerRefFuture.bad` so we still exhibit the same bad result if I make it an explicit ref.

> make the compiler error/thwart kernel generation with ref intents on scalars?

Should be fairly low effort. For reduction intents we just make it not gpuizable (look in `symsInBodyAreGpuizable`).
We could do a similar check in the same place (and probably have it error out rather than fail to gpuize). Just look for syms flagged `FLAG_TASK_PRIVATE_VARIABLE` and error out if they're scalars.